src/controller_examples/zookeeper_controller/exec/reconciler.rs

```diff
@@ -166,27 +166,75 @@ pub fn reconcile_core(
             return (state_prime, req_o);
         },
         ZookeeperReconcileStep::AfterCreateConfigMap => {
-            let stateful_set = make_stateful_set(zk);
-            let req_o = Option::Some(KubeAPIRequest::CreateRequest(
-                KubeCreateRequest {
+            let req_o = Option::Some(KubeAPIRequest::GetRequest(
+                KubeGetRequest {
                     api_resource: StatefulSet::api_resource(),
+                    name: zk.name().unwrap(),
                     namespace: zk.namespace().unwrap(),
-                    obj: stateful_set.to_dynamic_object(),
                 }
             ));
             let state_prime = ZookeeperReconcileState {
-                reconcile_step: ZookeeperReconcileStep::Done,
+                reconcile_step: ZookeeperReconcileStep::AfterGetStatefulSet,
                 ..state
             };
             return (state_prime, req_o);
         },
+        ZookeeperReconcileStep::AfterGetStatefulSet => {
+            if resp_o.is_some() && resp_o.as_ref().unwrap().is_get_response() {
+                let stateful_set = make_stateful_set(zk);
+                let get_sts_resp = resp_o.unwrap().into_get_response().res;
+                if get_sts_resp.is_ok() {
+                    // update
+                    let old_sts_o = get_sts_resp.unwrap();
+                    let update_sts = StatefulSet::from_dynamic_object(old_sts_o);
```

Comment:
Can you rename this `update_sts` to `found_sts`? And there is no need to define `old_sts_o` since you only use it once here.